cpp/open3d/t/geometry/kernel/VtkUtils.cpp

```diff
@@ -0,0 +1,240 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// The MIT License (MIT)
+//
+// Copyright (c) 2018-2021 www.open3d.org
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#include "open3d/t/geometry/kernel/VtkUtils.h"
+
+#include "vtkArrayDispatch.h"
+#include "vtkCellArray.h"
+#include "vtkDoubleArray.h"
+#include "vtkFloatArray.h"
+#include "vtkPoints.h"
+
+namespace open3d {
+namespace t {
+namespace geometry {
+namespace kernel {
+namespace vtkutils {
+
+vtkSmartPointer<vtkPoints> CreateVtkPointsFromTensor(
+        const core::Tensor& tensor) {
+    core::AssertTensorShape(tensor, {utility::nullopt, 3});
+    core::AssertTensorDtypes(tensor, {core::Float32, core::Float64});
+
+    auto tensor_cpu = tensor.To(core::Device()).Contiguous();
+    const size_t num_points = tensor.GetLength();
+
+    vtkSmartPointer<vtkPoints> pts = vtkSmartPointer<vtkPoints>::New();
+
+    if (tensor.GetDtype() == core::Float32) {
+        pts->SetDataTypeToFloat();
+        pts->SetNumberOfPoints(num_points);
+        const float* data = tensor_cpu.GetDataPtr<float>();
+        for (size_t i = 0; i < num_points; ++i) {
```

Comment:
Is there away to create vtk points from a buffer?

Reply:
I did some more research on this. It seems possible but we need to use a different array interface.
I think it is better to create a separate PR for this.